cogs/make_applicant.py

```diff
@@ -51,15 +51,20 @@ async def _perform_make_applicant(self, ctx: TeXBotApplicationContext, applicant
 
         AUDIT_MESSAGE: Final[str] = f"{ctx.user} used TeX Bot Command \"Make User Applicant\""
 
-        await applicant_member.add_roles(applicant_role, reason=AUDIT_MESSAGE)
-
-        logger.debug("Applicant role given to user %s", applicant_member)
-
         if guest_role in applicant_member.roles:
             await applicant_member.remove_roles(guest_role, reason=AUDIT_MESSAGE)
             logger.debug("Removed Guest role from user %s", applicant_member)
 
 
+        if applicant_role not in applicant_member.roles:
```

Comment:
What is the need for checking if they already have the role (other than an improved response message). Is an error raised if we attempt to give a user a role that they already have?

Reply:
No error - no, it's just to bring it into consistency with what the induct command does, and also saves the overhead of calling the add_role method when we don't need to

Reply:
Agreed that it should only be added if they don't have the role. I think the better check would to check if the role *is* within the member's roles. If it already is then do an fail fast and state that they are already applicant. Then go on to add the applicant role otherwise.

Should their welcome message have a reaction applied if they already were an applicant? (/gen) I don't think so.

Reply:
Excellent points - will fix